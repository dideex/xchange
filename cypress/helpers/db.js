const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
