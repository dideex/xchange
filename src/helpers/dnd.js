import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
