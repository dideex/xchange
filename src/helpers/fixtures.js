export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}