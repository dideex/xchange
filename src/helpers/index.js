export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
