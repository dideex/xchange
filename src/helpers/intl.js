import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
