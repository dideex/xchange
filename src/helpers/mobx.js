import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
