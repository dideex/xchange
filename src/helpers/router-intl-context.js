import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
