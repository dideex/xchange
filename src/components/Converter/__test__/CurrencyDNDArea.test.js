import React from 'react'
import Component from '../CurrencyDNDArea'
import {mountWithIntl} from '../../../helpers/intl'
import {fakeCurrnecy} from '../../../helpers/fixtures'

describe('Converter: Currency dnd area', () => {
  const props = {
    currency: fakeCurrnecy,
    loading: false,
  }
  const coordinates = {
    x: 10,
    y: 5,
  }
  describe('Markup', () => {
    it('Basic markup', () => {
      const wrapper = mountWithIntl(<Component {...props} />)
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('Should render spinner when loading', () => {
      const wrapper = mountWithIntl(<Component currency={[]} loading />)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('Component behaviour', () => {
    it.only('Should handle coordinates on click', () => {
      const wrapper = mountWithIntl(<Component {...props} />)
      // console.log('TCL: wrapper', wrapper.debug())
      // console.log(wrapper.find('[data-testid="currency-badge-item"]').first().debug())
      wrapper
        .find('[data-testid="currency-badge-item"]')
        .first()
        .simulate('mousedown', {
          currentTarget: {
            getBoundingClientRect() {
              return {
                x: coordinates.x,
                y: coordinates.y,
              }
            },
          },
        })
      console.log(wrapper.find('CurrencyDNDArea').instance().state)
    })
  })
})
