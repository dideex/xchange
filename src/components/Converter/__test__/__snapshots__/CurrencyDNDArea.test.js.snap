// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter: Currency dnd area Markup Basic markup 1`] = `"<div class=\\"css-1lppr29\\"><div class=\\"css-adjgff\\"><svg viewBox=\\"0 0 38 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #aee239;\\" d=\\"M37.4832 32.8537L30.0829 25.4548C29.749 25.1208 29.2962 24.9353 28.8212 24.9353H27.6115C29.6599 22.3154 30.8771 19.0208 30.8771 15.436C30.8771 6.90902 23.9666 0 15.4389 0C6.91122 0 0 6.90902 0 15.436C0 23.963 6.91047 30.872 15.4389 30.872C19.0237 30.872 22.3197 29.6548 24.9396 27.6064V28.8162C24.9396 29.2912 25.1251 29.7439 25.4591 30.0779L32.8594 37.4768C33.5571 38.1744 34.6852 38.1744 35.3755 37.4768L37.4758 35.3764C38.1735 34.6787 38.1735 33.5513 37.4832 32.8537ZM15.4389 26.122C9.53558 26.122 4.75146 21.3468 4.75146 15.436C4.75146 9.5334 9.52741 4.75 15.4389 4.75C21.3422 4.75 26.1264 9.52523 26.1264 15.436C26.1264 21.3386 21.3504 26.122 15.4389 26.122ZM17.451 14.8007L14.1104 13.7987C13.7274 13.6837 13.4595 13.2955 13.4595 12.8539C13.4595 12.3144 13.8529 11.875 14.3353 11.875H16.4216C16.76 11.875 17.0866 11.9707 17.373 12.1511C17.6135 12.3018 17.9193 12.2929 18.1249 12.0969L18.9969 11.2649C19.2589 11.0148 19.2441 10.5814 18.9546 10.3639C18.2792 9.85699 17.4643 9.56457 16.6264 9.52152V8.3125C16.6264 7.98445 16.3607 7.71875 16.0327 7.71875H14.8452C14.5171 7.71875 14.2514 7.98445 14.2514 8.3125V9.50891C12.4976 9.55566 11.0838 11.0341 11.0838 12.8539C11.0838 14.3361 12.0479 15.6602 13.4276 16.0743L16.7682 17.0763C17.1511 17.1913 17.4191 17.5795 17.4191 18.0211C17.4191 18.5606 17.0257 19 16.5433 19H14.4577C14.1193 19 13.7927 18.9043 13.5063 18.7239C13.2658 18.5732 12.96 18.5821 12.7544 18.7781L11.8824 19.6101C11.6204 19.8602 11.6352 20.2936 11.9247 20.5111C12.6001 21.018 13.415 21.3104 14.2529 21.3535V22.5625C14.2529 22.8905 14.5186 23.1562 14.8466 23.1562H16.0341C16.3622 23.1562 16.6279 22.8905 16.6279 22.5625V21.3661C18.3817 21.3193 19.7955 19.8416 19.7955 18.0211C19.7955 16.5389 18.8314 15.2148 17.451 14.8007Z\\"></path></svg><input type=\\"text\\" placeholder=\\"Find\\" class=\\"css-487mdy\\" value=\\"\\"></div><div class=\\"css-eex5l9\\"><div class=\\"css-1i7yuyg\\"><div class=\\"css-17ixpwd\\"></div><div class=\\"css-17ixpwd\\"></div><div class=\\"css-17ixpwd\\"></div></div></div></div>"`;

exports[`Converter: Currency dnd area Markup Should render spinner when loading 1`] = `"<div class=\\"css-1lppr29\\"><div class=\\"css-adjgff\\"><svg viewBox=\\"0 0 38 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #aee239;\\" d=\\"M37.4832 32.8537L30.0829 25.4548C29.749 25.1208 29.2962 24.9353 28.8212 24.9353H27.6115C29.6599 22.3154 30.8771 19.0208 30.8771 15.436C30.8771 6.90902 23.9666 0 15.4389 0C6.91122 0 0 6.90902 0 15.436C0 23.963 6.91047 30.872 15.4389 30.872C19.0237 30.872 22.3197 29.6548 24.9396 27.6064V28.8162C24.9396 29.2912 25.1251 29.7439 25.4591 30.0779L32.8594 37.4768C33.5571 38.1744 34.6852 38.1744 35.3755 37.4768L37.4758 35.3764C38.1735 34.6787 38.1735 33.5513 37.4832 32.8537ZM15.4389 26.122C9.53558 26.122 4.75146 21.3468 4.75146 15.436C4.75146 9.5334 9.52741 4.75 15.4389 4.75C21.3422 4.75 26.1264 9.52523 26.1264 15.436C26.1264 21.3386 21.3504 26.122 15.4389 26.122ZM17.451 14.8007L14.1104 13.7987C13.7274 13.6837 13.4595 13.2955 13.4595 12.8539C13.4595 12.3144 13.8529 11.875 14.3353 11.875H16.4216C16.76 11.875 17.0866 11.9707 17.373 12.1511C17.6135 12.3018 17.9193 12.2929 18.1249 12.0969L18.9969 11.2649C19.2589 11.0148 19.2441 10.5814 18.9546 10.3639C18.2792 9.85699 17.4643 9.56457 16.6264 9.52152V8.3125C16.6264 7.98445 16.3607 7.71875 16.0327 7.71875H14.8452C14.5171 7.71875 14.2514 7.98445 14.2514 8.3125V9.50891C12.4976 9.55566 11.0838 11.0341 11.0838 12.8539C11.0838 14.3361 12.0479 15.6602 13.4276 16.0743L16.7682 17.0763C17.1511 17.1913 17.4191 17.5795 17.4191 18.0211C17.4191 18.5606 17.0257 19 16.5433 19H14.4577C14.1193 19 13.7927 18.9043 13.5063 18.7239C13.2658 18.5732 12.96 18.5821 12.7544 18.7781L11.8824 19.6101C11.6204 19.8602 11.6352 20.2936 11.9247 20.5111C12.6001 21.018 13.415 21.3104 14.2529 21.3535V22.5625C14.2529 22.8905 14.5186 23.1562 14.8466 23.1562H16.0341C16.3622 23.1562 16.6279 22.8905 16.6279 22.5625V21.3661C18.3817 21.3193 19.7955 19.8416 19.7955 18.0211C19.7955 16.5389 18.8314 15.2148 17.451 14.8007Z\\"></path></svg><input type=\\"text\\" placeholder=\\"Find\\" class=\\"css-487mdy\\" value=\\"\\"></div><div width=\\"100%\\" height=\\"200\\" class=\\"css-fnrdyw\\"><svg class=\\"loading\\" width=\\"96\\" height=\\"96\\" viewBox=\\"0 0 256 256\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path class=\\"clockwise-big\\" d=\\"M223.416 97.9999C218.569 82.6432 210.093 68.6803 198.707 57.2934C187.32 45.9065 173.357 37.4308 158 32.5839V3.56995C180.899 9.10903 201.828 20.8534 218.487 37.5126C235.147 54.1717 246.891 75.1008 252.43 97.9999H223.416ZM98 32.5839C82.6431 37.4305 68.68 45.9061 57.2931 57.293C45.9062 68.68 37.4306 82.643 32.584 97.9999H3.57001C9.10888 75.1007 20.8532 54.1715 37.5124 37.5123C54.1715 20.8531 75.1007 9.10882 98 3.56995V32.5839ZM32.584 158C37.4309 173.357 45.9066 187.32 57.2935 198.706C68.6804 210.093 82.6432 218.569 98 223.416V252.43C75.1008 246.891 54.1717 235.146 37.5126 218.487C20.8535 201.828 9.10909 180.899 3.57001 158H32.584ZM158 223.416C173.357 218.569 187.319 210.093 198.706 198.706C210.093 187.319 218.569 173.357 223.416 158H252.43C246.89 180.899 235.146 201.828 218.487 218.487C201.828 235.146 180.899 246.89 158 252.43V223.416Z\\" style=\\"fill: #40c0cb;\\"></path><path class=\\"counter-clockwise\\" d=\\"M193.942 107.266C190.592 96.6555 184.735 87.0081 176.868 79.1403C169 71.2725 159.353 65.4159 148.742 62.066V42C164.568 45.8289 179.032 53.9465 190.545 65.4606C202.057 76.9747 210.173 91.4397 214 107.266H193.942ZM107.266 62.066C96.6555 65.4159 87.0081 71.2725 79.1403 79.1403C71.2725 87.0081 65.4159 96.6555 62.066 107.266H42C45.8276 91.4387 53.9446 76.973 65.4588 65.4588C76.973 53.9446 91.4387 45.8276 107.266 42V62.058V62.066ZM62.066 148.742C65.4159 159.353 71.2725 169 79.1403 176.868C87.0081 184.735 96.6555 190.592 107.266 193.942V214C91.4387 210.172 76.973 202.055 65.4588 190.541C53.9446 179.027 45.8276 164.561 42 148.734H62.058L62.066 148.742ZM148.742 193.942C159.353 190.592 169 184.735 176.868 176.868C184.735 169 190.592 159.353 193.942 148.742H214C210.171 164.568 202.054 179.032 190.539 190.545C179.025 202.057 164.56 210.173 148.734 214V193.942H148.742Z\\" style=\\"fill: #40c0cb;\\"></path></svg></div></div>"`;
